common/flatpak-prune.c

```diff
@@ -381,10 +381,10 @@ flatpak_ostree_object_name_hash (gconstpointer a)
      those are the ones that will be first compared on a hash collision,
      so if they were always the same that would waste 4 comparisons. */
   return
-    data[32] |
-    data[31] << 8 |
-    data[30] << 16 |
-    data[29] << 24;
+    ((guint32) data[32]) |
+    ((guint32) data[31]) << 8 |
+    ((guint32) data[30]) << 16 |
+    ((guint32) data[29]) << 24;
```

Comment:
Doesn't it have to be `((guint8) data[32])`?

Reply:
no. you do the type promotion first and then bitshift. Otherwise it's indeed UB.

Reply:
Alright